tools/src/main/scala/scala/scalanative/optimizer/pass/MethodLowering.scala

```diff
@@ -37,10 +38,12 @@ class MethodLowering(implicit fresh: Fresh, top: Top) extends Pass {
         val typeptr = let(Op.Load(Type.Ptr, obj))
         val idptr   = let(Op.Elem(Rt.Type, typeptr, Seq(Val.Int(0), Val.Int(0))))
         val id      = let(Op.Load(Type.Int, idptr))
-        val methptrptr = let(
-          Op.Elem(top.dispatchTy,
-                  top.dispatchVal,
-                  Seq(Val.Int(0), id, Val.Int(meth.id))))
+        val rowptr = let(
+          Op.Elem(Type.Ptr,
+                  top.tables.dispatchVal,
+                  Seq(Val.Int(top.tables.dispatchOffset(meth.id)))))
+        val methptrptr =
+          let(Op.Elem(Type.Ptr, rowptr, Seq(id)))
```

Comment:
Although this adds one more instruction, first one is guaranteed to be constant-folded in LLVM optimizer (all arguments of the instructions are statically-known values and the instruction is side-effect-free.)